syntax = "proto3";

service Worker {
    rpc SetupAPI(Empty) returns (Empty) {}
    rpc Exec(Code) returns (Empty) {}
    rpc Silence(Empty) returns (Empty) {}
    rpc ClearBuffer(Empty) returns (Empty) {}
    rpc SetNumInputs(NumChannels) returns (Empty) {}
    rpc SetNumOutputs(NumChannels) returns (Empty) {}
    rpc GetNumInputs(Empty) returns (NumChannels) {}
    rpc GetNumOutputs(Empty) returns (NumChannels) {}
    rpc GetSampleRate(Empty) returns (SampleRate) {}
    rpc GetBufferSize(Empty) returns (BufferSize) {}
}

message Code {
    string code = 1;
}

message NumChannels {
    uint32 num = 1;
}

message BufferSize {
    uint32 size = 1;
}

message SampleRate {
    float rate = 1;
}

message Empty {

}
